extension/android/CMakeLists.txt

```diff
@@ -32,9 +49,11 @@ if(CMAKE_TOOLCHAIN_ANDROID)
   target_compile_options(executorch_jni PUBLIC ${_common_compile_options})
 
   add_library(executorch_llama_jni SHARED jni/jni_layer_llama.cpp)
-  target_link_libraries(executorch_llama_jni fbjni llama_runner portable_ops_lib)
+  target_link_libraries(executorch_llama_jni extension_data_loader
+                        extension_module fbjni llama_runner "-Wl,--whole-archive" portable_ops_lib executorch
+                        portable_kernels "-Wl,--no-whole-archive" custom_ops)
   if(EXECUTORCH_BUILD_XNNPACK)
-    target_link_libraries(executorch_llama_jni xnnpack_backend)
+    target_link_libraries(executorch_llama_jni "-Wl,--whole-archive"  "-Wl,--no-whole-archive"  xnnpack_backend cpuinfo XNNPACK pthreadpool)
```

Comment:
xnnpack out of whole-archive, causing missing backend registration

Reply:
--whole-archive --no-whole-archive should surround only xnnpack_backend? It should be 

```
target_link_libraries(executorch_llama_jni "-Wl,--whole-archive" xnnpack_backend "-Wl,--no-whole-archive" cpuinfo XNNPACK pthreadpool)
```